client/dashboard/data/site-plans.ts

```diff
@@ -1,9 +1,10 @@
 import wpcom from 'calypso/lib/wp';
 
 export interface Plan {
-	id: string | null;
+	id?: string | null;
```

Comment:
Question: so `null` and `undefined` (not present) is different here?

Reply:
Yup 😅 Perhaps I should document the cases I've found.

`id` looks like it's only present when `current_plan` is true.
For free plans the `id === null`. Fair enough, there's no purchase to line up with
For a non-free, current plan, that is when `id` is a string.

Reply:
Thanks for the explanation! Perhaps it'll be *slightly* clearer if the interface name is `SitePlan` instead, because it's really tied to the current site 🤔

Reply:
Yes, however unfortunately there is already a `SitePlan`. 🙃 

The `Site` object has a `plan` field, and its format is slightly different.

So `SitePlan` I think gets to be called `SitePlan` because it is a child component of the `Site`. So I'm not quite sure what to call _this_ plan object.

I suppose ideally the fix would be to refactor the backend and ensure everything is actually using the exact same format across everything!